### PR TITLE
[imageIO] ask "Linear" colorspace for RAW images

### PR DESCRIPTION
**Pull request by fabiencastan**
Follow the changes in oiio to manage colorspaces on RAW images, see: https://github.com/OpenImageIO/oiio/pull/2260

And deal with compatibility with previous oiio versions.

### CODE REVIEW

File: src/aliceVision/image/io.cpp [hunk 1]
```diff
@@ -164,17 +165,40 @@ void readImage(const std::string& path,
...
+#endif
 
   oiio::ImageBuf inBuf(path, 0, 0, NULL, &configSpec);
 
   inBuf.read(0, 0, true, oiio::TypeDesc::FLOAT); // force image convertion to float (for grayscale and color space convertion)
 
-  if(!inBuf.initialized())
-    throw std::runtime_error("Can't find/open image file '" + path + "'.");
+  if (!inBuf.initialized())
+      throw std::runtime_error("Can't find/open image file '" + path + "'.");
```
Comment (simogasp) [inline src/aliceVision/image/io.cpp]: Cannot

File: src/aliceVision/image/io.cpp [hunk 2]
```diff
@@ -164,17 +165,40 @@ void readImage(const std::string& path,
...
 
+#if OIIO_VERSION <= (10000 * 2 + 100 * 0 + 8) // OIIO_VERSION <= 2.0.8
+  // Workaround for bug in RAW colorspace management in previous versions of OIIO:
+  //     When asking sRGB we got sRGB primaries with linear gamma,
+  //     but oiio::ColorSpace was wrongly set to sRGB.
+  oiio::ImageSpec inSpec = inBuf.spec();
+  if(inSpec.get_string_attribute("oiio:ColorSpace", "") == "sRGB")
+  {
+      auto in = oiio::ImageInput::open(path, nullptr);
```
Comment (simogasp) [inline src/aliceVision/image/io.cpp]: can be const?

File: src/aliceVision/image/io.cpp [hunk 3]
```diff
@@ -164,17 +165,40 @@ void readImage(const std::string& path,
...
+  //     but oiio::ColorSpace was wrongly set to sRGB.
+  oiio::ImageSpec inSpec = inBuf.spec();
+  if(inSpec.get_string_attribute("oiio:ColorSpace", "") == "sRGB")
+  {
+    const auto in = oiio::ImageInput::open(path, nullptr);
+    const std::string formatStr = in->format_name();
+    if(formatStr == "raw")
+    {
+      // For the RAW plugin: override colorspace as linear (as the content is linear with sRGB primaries but declared as sRGB)
+      configSpec.attribute("oiio:ColorSpace", "Linear");
```
Comment (yann-lty) [inline src/aliceVision/image/io.cpp]: inSpec

File: src/aliceVision/imageIO/image.cpp [hunk 4]
```diff
@@ -127,35 +128,66 @@ void readImage(const std::string& path,
...
+    //     but oiio::ColorSpace was wrongly set to sRGB.
+    oiio::ImageSpec inSpec = inBuf.spec();
+    if(inSpec.get_string_attribute("oiio:ColorSpace", "") == "sRGB")
+    {
+        const auto in = oiio::ImageInput::open(path, nullptr);
+        const std::string formatStr = in->format_name();
+        if(formatStr == "raw")
+        {
+            // For the RAW plugin: override colorspace as linear (as the content is linear with sRGB primaries but declared as sRGB)
+            configSpec.attribute("oiio:ColorSpace", "Linear");
```
Comment (yann-lty) [inline src/aliceVision/imageIO/image.cpp]: inSpec

File: src/aliceVision/image/io.cpp (hunk 3 above)
Comment (fabiencastan) [reply src/aliceVision/image/io.cpp]: oops

**Pull request merged**